web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\hoeringsportal_citizen_proposal\Helper;
+
+use Drupal\Core\State\State;
+use Drupal\Core\TempStore\PrivateTempStoreFactory;
+use Drupal\Core\Url;
+use Drupal\node\Entity\Node;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Serializer\Serializer;
+use Drupal\Core\Messenger\MessengerTrait;
+
+/**
+ * A helper class for the module.
+ */
+class Helper {
+  use MessengerTrait;
+
+  /**
+   * Constructor for the citizen proposal helper class.
+   */
+  public function __construct(
+    readonly private PrivateTempStoreFactory $tempStoreFactory,
+    readonly private Serializer $serializer,
+    readonly private State $state,
+  ) {
+  }
+
+  /**
+   * Validate that the tempstore contains a node and return it.
+   *
+   * @return object|null
+   *   The entity or NULL if no valid entity was found.
+   */
+  public function tempStoreValid(): ?object {
+    $entitySerialized = $this->tempStoreFactory->get('hoeringsportal_citizen_proposal')->get('citizen_proposal_entity');
+    $node = is_string($entitySerialized) ? $this->serializer->deserialize($entitySerialized, Node::class, 'json') : NULL;
+
+    return isset($node) && $node instanceof (Node::class) ? $node : NULL;
+  }
+
+  /**
+   * Delete the citizen_proposal_entity tempstore.
+   */
+  public function tempStoreDelete(): void {
+    try {
+      $this->tempStoreFactory->get('hoeringsportal_citizen_proposal')->delete('citizen_proposal_entity');
```

Comment:
The expression `$this->tempStoreFactory->get('hoeringsportal_citizen_proposal')` is used multiple times throgout this class. Consider adding a helper method to get it:

```php
private function getProposalStorage(): {
return $this->tempStoreFactory->get('hoeringsportal_citizen_proposal');
}
```

Furthermore, the key `citizen_proposal_entity` is also used multiple times. Consider adding it as a `const`:

```php
private const CITIZEN_PROPOSAL_ENTITY = 'citizen_proposal_entity';

$this->getProposalStorage()->delete(self::CITIZEN_PROPOSAL_ENTITY);
…
```

Reply:
Added private method and constant.